app/assets/javascripts/initializers/initializeArticleReactions.js

```diff
@@ -40,12 +40,15 @@ function hasUserReacted(reactionName) {
 }
 
 function getNumReactions(reactionName) {
-  var num = document.getElementById('reaction-number-' + reactionName)
-    .textContent;
-  if (num === '') {
+  const reactionEl = document.getElementById('reaction-number-' + reactionName);
+
+  if (!reactionEl) {
```

Comment:
Looking at these functions, if we have this code:
```
if (!reactionEl) {
    return;
 }
```

and we're using `getNumReactions` in the following way from `reactToArticle`:
```
function reactToArticle(articleId, reaction) {
  // Visually toggle the reaction
  function toggleReaction() {
    var currentNum = getNumReactions(reaction);
    if (hasUserReacted(reaction)) {
      hideUserReaction(reaction);
      setReactionCount(reaction, currentNum - 1);
    } else {
      showUserReaction(reaction, 'user-animated');
      setReactionCount(reaction, currentNum + 1);
    }
  }
```
currentNum seems like it would be undefined because we return early if !reactionEl, hence won't it cause the following evaluation `undefined - 1`?

Reply:
Yeah, I was wondering to myself whether we want to allow the error to bubble up to `reactToArticle`, or whether we just want `getNumReactions` to return `0` if there isn't a `reactionEl ` (to keep the math working). I think probably the latter makes sense here, now that I look at the code again.

Reply:
@Ridhwana Updated this to just return `0` in the latest commit, which is what we do if the `reactionEl`'s `textContent` is an empty string in any case 🙃